### PR TITLE
github actions

### PR DESCRIPTION
**Pull request by matbesancon**
adding github action for CI and code coverage

### CODE REVIEW

**Comment by matbesancon** [comment]
ping @joaquimg

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        version:
+          - '1.0'
+          - '1'
+          - 'nightly'
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
+        arch:
+          - x64
```
Comment (blegat) [inline .github/workflows/ci.yml]: Might be more useful to test for 64 and 32-bit (to catch the `Int32` vs `Int64` common gotcha's) but only on linux.

Comment (matbesancon) [reply .github/workflows/ci.yml]: done, I took the one from MOI but added OSX

**Review by joaquimg — APPROVED** [review]
Looks good!
I would not add mac and win because of CI time. But if it is fast should be ok.

**Comment by matbesancon** [comment]
I think CI is fairly fast on GH-actions. Worst thing we can remove them later

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            arch: x64
+          - version: '1.0'
+            os: ubuntu-latest
+            arch: x86
+          - version: '1'
+            os: windows-latest
+            arch: x64
+          - version: '1'
+            os: macOS-latest
+            arch: x64
```
Comment (odow) [inline .github/workflows/ci.yml]: The rule-of-thumb I went with was:
- Does the package have binary dependencies?
  - If yes, test on Linux, Mac, and Window 64-bit, Julia 1 and Julia 1.0
    - Also test on linux 32-bit Julia 1.0 for `Int32` mistakes
    - https://github.com/jump-dev/GLPK.jl/blob/master/.github/workflows/ci.yml
  - If no, test on Linux 64-bit, Julia 1 and 1.0, and Linux 32-bit, Julia 1.0
    - Optionally test on Window 64-bit, Julia 1.0 to test windows installation
    - https://github.com/jump-dev/MathOptInterface.jl/blob/master/.github/workflows/ci.yml

**Comment by matbesancon** [comment]
Since MOI has binary dependencies, then all depending packages also have
some, or do you mean direct dependencies?

**Comment by odow** [comment]
Usually just direct. You can assume the down-stream dependencies are well-tested, so you just want to test the first-order integration with binaries (e.g., solvers).

File: .github/workflows/ci.yml (hunk 2 above)
Comment (matbesancon) [reply .github/workflows/ci.yml]: recommendation followed in last commit

**Review by blegat — APPROVED** [review]

**Comment by matbesancon** [comment]
ping @joaquimg for merging this one :)

**Comment by joaquimg** [comment]
Nice!

**Pull request merged**